### PR TITLE
PULSEBS-1: As a student I want to book a seat for one of my lectures so that I can attend it

### PR DESCRIPTION
**Pull request by enricocarraro**
Notes:
- I changed the configuration such that requests made by the front-end are redirected to the back-end, so no need to implement make requests explicitly (e.g. `axios.post("http://localhost:4001/ticket"...` can be written as `axios.post("ticket"...`);
- In the back-end server I used the URL prefix `api/` for routes.

### CODE REVIEW

File: server/controllers/user-controller.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            httpOnly: true,
+            sameSite: true,
+            maxAge: 1000 * expireTime,
+          });
+
+          res.json({
+            id: userDetails.id,
+            email: userDetails.email,
+            role: userDetails.role,
+          });
```
Comment (280113) [inline server/controllers/user-controller.js]: Consider to send to the client the information always in the same way

**Review by 280113 — COMMENTED** [review]
With respect to the db (https://github.com/giovannitangredi/PULSeBS/blob/1623a6fe777dca50c92f833b23f5bdc6809c64a7/server/db.js) it could be convenient to have a table to link students/teachers to the courses they attend

File: server/controllers/user-controller.js (hunk 1 above)
Comment (enricocarraro) [reply server/controllers/user-controller.js]: Resolved.

**Comment by enricocarraro** [comment]
> With respect to the db (https://github.com/giovannitangredi/PULSeBS/blob/1623a6fe777dca50c92f833b23f5bdc6809c64a7/server/db.js) it could be convenient to have a table to link students/teachers to the courses they attend

@280113, I think this is already possible using table `lecture` and `lecture_booking`, since they contain both information on the students who attended a particular lecture and who taught it. 
https://github.com/giovannitangredi/PULSeBS/blob/0b965bfd35af14c0dda51b5a77a83f7cad1bfea2/server/db.js#L70-L94

**Pull request merged**